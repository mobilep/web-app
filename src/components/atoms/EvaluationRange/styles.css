.EvaluationRange {
	border-radius: 18px;
	box-shadow: 0 0 25px rgba(0, 0, 0, .15);
	display: flex;
	padding: 0 5px;
}

.EvaluationRange-evaluation {
	position: relative;
	flex: 0 0 58px;
}

.EvaluationRange-evaluation input {
	left: 0;
	opacity: 0;
	position: absolute;
	top: 0;
}

.EvaluationRange-evaluation input + label {
	width: 100%;
	border-radius: 18px;
	cursor: pointer;
	margin: 0;
	padding: 5px 0;
	text-align: center;
}

.EvaluationRange-evaluation input:checked + label {
	position: absolute;
	left: 50%;
	top: 0;
	width: 70px;
	height: 100%;
	transform: translateX(-50%);
	background-color: var(--tomatoRed);
	color: white;
}

.EvaluationRange-evaluation input:focus + label {
	outline: 5px auto -webkit-focus-ring-color;
}

@media (max-width: 345px) {
	.EvaluationRange-evaluation {
		flex: 0 0 48px;
	}
}
