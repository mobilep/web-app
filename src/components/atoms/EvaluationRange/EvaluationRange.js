import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './styles.css';

export default class EvaluationRange extends Component {

	state = {
		value: 0,
	}

	handleChange = (event) => {
		const { onChange } = this.props;
		const { value } = event.target;

		this.setState({ value: +value });
		onChange(event);
	}

	getEvaluationsList = (start, end) => {
		const listLength = Math.abs(end - start) + 1;
		return Array(listLength).fill().map((e, i) => (start < end) ? (start + i) : (start - i));
	}

	render () {
		const { value } = this.state;
		const {
			start,
			end,
			name,
			disabled,
			required,
		} = this.props;

		const propValue = this.props.value;

		return (
			<div className='EvaluationRange'>
				{this.getEvaluationsList(start, end).map((evaluation) => (
					<div className='EvaluationRange-evaluation' key={evaluation}>
						<input
							required={required}
							type='radio'
							name={name}
							id={`${name}-${evaluation}`}
							checked={propValue && disabled
								? propValue === evaluation
								: value === evaluation}
							onChange={this.handleChange}
							value={evaluation}
							disabled={disabled}
						/>
						<label htmlFor={`${name}-${evaluation}`}>{evaluation}</label>
					</div>
				))}
			</div>
		);
	}
}

EvaluationRange.defaultProps = {
	start: 1,
	end: 5,
	onChange: () => {},
	name: 'evaluation',
};

EvaluationRange.propTypes = {
	disabled: PropTypes.bool,
	end: PropTypes.number,
	name: PropTypes.string,
	onChange: PropTypes.func,
	required: PropTypes.bool,
	start: PropTypes.number,
	value: PropTypes.number,
};
