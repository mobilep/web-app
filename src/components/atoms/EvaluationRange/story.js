import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { number } from '@storybook/addon-knobs/react';
import EvaluationRange from './EvaluationRange';

storiesOf('1. Atoms / EvaluationRange', module)
	.add('Evaluation Range', () => (
		<EvaluationRange
			start={number('start', 1)}
			end={number('end', 5)}
			value={number('value', 3)}
			onChange={action('onChange')}
		/>
	));
