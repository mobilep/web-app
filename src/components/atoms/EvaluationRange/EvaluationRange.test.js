import React from 'react';
import { shallow } from 'enzyme';

import EvaluationRange from './EvaluationRange';

describe('Atom - EvaluationRange', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<EvaluationRange />
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render with value', () => {
		const wrapper = shallow(
			<EvaluationRange value={2} />
		);
		expect(wrapper).toMatchSnapshot();
	});

});

