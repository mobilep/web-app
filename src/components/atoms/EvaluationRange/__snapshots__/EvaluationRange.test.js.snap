// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - EvaluationRange Should render 1`] = `
<div
  className="EvaluationRange"
>
  <div
    className="EvaluationRange-evaluation"
    key="1"
  >
    <input
      checked={false}
      id="evaluation-1"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={1}
    />
    <label
      htmlFor="evaluation-1"
    >
      1
    </label>
  </div>
  <div
    className="EvaluationRange-evaluation"
    key="2"
  >
    <input
      checked={false}
      id="evaluation-2"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={2}
    />
    <label
      htmlFor="evaluation-2"
    >
      2
    </label>
  </div>
  <div
    className="EvaluationRange-evaluation"
    key="3"
  >
    <input
      checked={false}
      id="evaluation-3"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={3}
    />
    <label
      htmlFor="evaluation-3"
    >
      3
    </label>
  </div>
  <div
    className="EvaluationRange-evaluation"
    key="4"
  >
    <input
      checked={false}
      id="evaluation-4"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={4}
    />
    <label
      htmlFor="evaluation-4"
    >
      4
    </label>
  </div>
  <div
    className="EvaluationRange-evaluation"
    key="5"
  >
    <input
      checked={false}
      id="evaluation-5"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={5}
    />
    <label
      htmlFor="evaluation-5"
    >
      5
    </label>
  </div>
</div>
`;

exports[`Atom - EvaluationRange Should render with value 1`] = `
<div
  className="EvaluationRange"
>
  <div
    className="EvaluationRange-evaluation"
    key="1"
  >
    <input
      checked={false}
      id="evaluation-1"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={1}
    />
    <label
      htmlFor="evaluation-1"
    >
      1
    </label>
  </div>
  <div
    className="EvaluationRange-evaluation"
    key="2"
  >
    <input
      checked={false}
      id="evaluation-2"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={2}
    />
    <label
      htmlFor="evaluation-2"
    >
      2
    </label>
  </div>
  <div
    className="EvaluationRange-evaluation"
    key="3"
  >
    <input
      checked={false}
      id="evaluation-3"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={3}
    />
    <label
      htmlFor="evaluation-3"
    >
      3
    </label>
  </div>
  <div
    className="EvaluationRange-evaluation"
    key="4"
  >
    <input
      checked={false}
      id="evaluation-4"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={4}
    />
    <label
      htmlFor="evaluation-4"
    >
      4
    </label>
  </div>
  <div
    className="EvaluationRange-evaluation"
    key="5"
  >
    <input
      checked={false}
      id="evaluation-5"
      name="evaluation"
      onChange={[Function]}
      type="radio"
      value={5}
    />
    <label
      htmlFor="evaluation-5"
    >
      5
    </label>
  </div>
</div>
`;
